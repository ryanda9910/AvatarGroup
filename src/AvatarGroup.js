/* eslint-disable jsx-a11y/img-redundant-alt */
import React from "react";
import "./Avatar.css";
import { avatars } from "./avatarData";


const avatarSizes = { xs: "2rem", sm: "3rem", md: "4rem", lg: "6rem" };

const Avatar = ({ name, size, image, numExtra, zIndex }) => {
  const initials = numExtra
    ? name
    : name
        .split(" ")
        .map((word) => word[0])
        .join("")
        .toUpperCase();

  return (
    <div
      style={{
        marginLeft: zIndex === 0 ? "unset" : `calc(-${avatarSizes[size]}/2.5)`,
        zIndex
      }}
      className={`avatar ${size}`}
    >
      {image ? <img src={image} alt={`${name}'s profile picture`} /> : initials}
    </div>
  );
};

const AvatarGroup = ({
  size = "md",
  maxLength = 4
}) => {
  const slicedAvatars = avatars.slice(0, maxLength);
  const numExtra = avatars.length - maxLength;

  return (
    <div className="avatar-list">
      {slicedAvatars.map((avatar, index) => (
        <Avatar
          key={index}
          name={avatar.name}
          size={size}
          image={avatar.image}
          zIndex={index}
        />
      ))}
      {numExtra > 0 && (
        <Avatar
          key="extra"
          numExtra={numExtra > 0}
          name={`+${numExtra}`}
          size={size}
          zIndex={maxLength}
        />
      )}
    </div>
  );
};

export default AvatarGroup;
