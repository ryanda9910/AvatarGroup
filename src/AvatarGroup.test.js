import { render } from "@testing-library/react";
import AvatarGroup from "./AvatarGroup";
import React from "react";

describe("Avatar Group component", () => {
  test("default component should render properly", () => {
    const output = render(<AvatarGroup />);
    expect(output).toMatchSnapshot();
  });

  it("should display a custom maximum number of avatars", () => {
   
    const maxAvatars = 4;

    const { container } = render(
      <AvatarGroup  maxLength={maxAvatars} />
    );
    const avatarList = container.querySelector(".avatar-list");

    expect(avatarList.children.length - 1).toBe(maxAvatars);// Check if the avatar list has a maximum based on maxAvatars
  });

  it("should render expectation size avatars", () => {
    const expectSize = ["xs", "sm", "md", "lg"];
    let size;
    const { container } = render(<AvatarGroup size="xs" />);
    const avatarElements = container.querySelectorAll(".avatar");
    size = avatarElements[0].className.split(' ')[1]
    expect(expectSize.includes(size)).toBeTruthy() //if outside the expectSize will be false
  });
});
