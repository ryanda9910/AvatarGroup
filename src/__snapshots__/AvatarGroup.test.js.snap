// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Group component default component should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="avatar-list"
      >
        <div
          class="avatar md"
          style="z-index: 0;"
        >
          <img
            alt="John's profile picture"
            src="https://randomuser.me/api/portraits/men/1.jpg"
          />
        </div>
        <div
          class="avatar md"
          style="z-index: 1;"
        >
          <img
            alt="John's profile picture"
            src="https://randomuser.me/api/portraits/men/3.jpg"
          />
        </div>
        <div
          class="avatar md"
          style="z-index: 2;"
        >
          JD
        </div>
        <div
          class="avatar md"
          style="z-index: 3;"
        >
          <img
            alt="John's profile picture"
            src="https://randomuser.me/api/portraits/men/1.jpg"
          />
        </div>
        <div
          class="avatar md"
          style="z-index: 4;"
        >
          +8
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="avatar-list"
    >
      <div
        class="avatar md"
        style="z-index: 0;"
      >
        <img
          alt="John's profile picture"
          src="https://randomuser.me/api/portraits/men/1.jpg"
        />
      </div>
      <div
        class="avatar md"
        style="z-index: 1;"
      >
        <img
          alt="John's profile picture"
          src="https://randomuser.me/api/portraits/men/3.jpg"
        />
      </div>
      <div
        class="avatar md"
        style="z-index: 2;"
      >
        JD
      </div>
      <div
        class="avatar md"
        style="z-index: 3;"
      >
        <img
          alt="John's profile picture"
          src="https://randomuser.me/api/portraits/men/1.jpg"
        />
      </div>
      <div
        class="avatar md"
        style="z-index: 4;"
      >
        +8
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
